### PR TITLE
Json api dep feature

### PR DESCRIPTION
**Pull request by tomnis**
- gson should be in `api` scope
- only use strict locking for 2.12
- i'll publish a 4.0.0 after this

### CODE REVIEW

**Comment by tomnis** [comment]
Just noticed one other thing I gotta fix

File: README.md [hunk 1]
```diff
@@ -14,6 +14,20 @@ We believe engineers should reason more scientifically about code performance, a
 More detailed documentation can be found [here.](https://workday.github.io/warp-core)
 
 ## Getting Started
+
+### Important and Subtle:
+In order for `-PallScalaVersions` to work as expected, you must clone this into a non-default directory, ie a directory that is not equal to the repository name.
```
Comment (tomnis) [inline README.md:19]: alright this should be good to go, this stuff with the clone directory and subtle 2 gradle issues took some time to figure out

Comment (tomnis) [reply README.md:19]: @sarcasticsimba @rooeque @tsoppet  just making sure y'all are aware of this

Comment (sarcasticsimba) [reply README.md:19]: oh wow okay that's good to know

**Pull request merged**